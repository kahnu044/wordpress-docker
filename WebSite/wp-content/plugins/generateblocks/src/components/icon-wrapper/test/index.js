import { mount } from 'enzyme';
import IconWrapper from '../index';
import generalSvgs from '../../icon-picker/svgs-general';

const ChildrenComponent = () => ( <h2> { 'My children component' } </h2> );

describe( 'IconWrapper', () => {
	it( 'should render children elements', () => {
		const wrapper = mount(
			<IconWrapper>
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render left icon', () => {
		const wrapper = mount(
			<IconWrapper icon={ generalSvgs.clock.icon }>
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render right icon', () => {
		const wrapper = mount(
			<IconWrapper icon={ generalSvgs.clock.icon } direction={ 'right' }>
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should hide children elements', () => {
		const wrapper = mount(
			<IconWrapper icon={ generalSvgs.clock.icon } hideChildren={ true }>
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render multiple children', () => {
		const wrapper = mount(
			<IconWrapper icon={ generalSvgs.clock.icon }>
				<ChildrenComponent />
				<ChildrenComponent />
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render without children', () => {
		const wrapper = mount(
			<IconWrapper icon={ generalSvgs.clock.icon } />
		);

		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'should render with wrapper element', () => {
		const wrapper = mount(
			<IconWrapper
				icon={ generalSvgs.clock.icon }
				hideWrapper={ false }
				wrapperClassname={ 'my-wrapper-class' }
			>
				<ChildrenComponent />
			</IconWrapper>
		);

		expect( wrapper ).toMatchSnapshot();
	} );
} );
