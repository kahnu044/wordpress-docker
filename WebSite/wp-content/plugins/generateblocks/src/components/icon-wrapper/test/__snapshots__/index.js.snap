// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWrapper should hide children elements 1`] = `
<_default
  hideChildren={true}
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
</_default>
`;

exports[`IconWrapper should render children elements 1`] = `
<_default>
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
</_default>
`;

exports[`IconWrapper should render left icon 1`] = `
<_default
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
</_default>
`;

exports[`IconWrapper should render multiple children 1`] = `
<_default
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
</_default>
`;

exports[`IconWrapper should render right icon 1`] = `
<_default
  direction="right"
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
>
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
</_default>
`;

exports[`IconWrapper should render with wrapper element 1`] = `
<_default
  hideWrapper={false}
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
  wrapperClassname="my-wrapper-class"
>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
  <ChildrenComponent>
    <h2>
       
      My children component
       
    </h2>
  </ChildrenComponent>
</_default>
`;

exports[`IconWrapper should render without children 1`] = `
<_default
  icon={
    <svg
      aria-hidden="true"
      height="1em"
      role="img"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
        fill="currentColor"
      />
    </svg>
  }
>
  <span
    className="gb-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": <svg
          aria-hidden="true"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm92.49 313l-20 25a16 16 0 01-22.49 2.5l-67-49.72a40 40 0 01-15-31.23V112a16 16 0 0116-16h32a16 16 0 0116 16v144l58 42.5a16 16 0 012.49 22.5z"
            fill="currentColor"
          />
        </svg>,
      }
    }
  />
</_default>
`;
